app/pages/admin.js

```diff
@@ -1,39 +1,25 @@
-import React, { useState, useEffect } from "react";
+import React, { useState} from "react";
 import Link from "next/link";
 
 export default function Admin() {
-  const [link, setLink] = useState({ name: "", link: "" });
-  const [links, setLinks] = useState(new Map());
 
-  const clickAddLink = () => addToMap(links.size + 1, link);
+  const [links, setLinks] = useState({});
+
+  const clickAddLink = () => addToMap();
+
   const deleteOnMap = (key) => {
     setLinks((prev) => {
       const newState = new Map(prev);
       newState.delete(key);
       return newState;
     });
   };
-  
- const resetLink = () =>
- {
-  setLink({ ...link, name: '' })
-  setLink({ ...link, link: '' })
- }
 
-  const getName = (key) =>
-  { 
-    let nameFromKey = links.get(key)
-    return nameFromKey.name
-  }
+ 
 
-  const getLink = (key) =>
-  { 
-    let linkFromKey = links.get(key)
-    return linkFromKey.link
-  }
-  const addToMap = (key, value) => {
-    setLinks((prev) => new Map([...prev, [key, value]]));
-  };
+  const addToMap = () => {
+    setLinks((prev) => ({...prev, _id: "4a5d6v", name: '', link: ''}))};
```

Comment:
Adding the attributes to the object. Was thinking more of doing  _id: "whatever" and linkObj with the contents {name: link:} as well.